### DISCUSSION TITLE
Bring window to current desktop

### DISCUSSION THREAD

**Discussion by andrepiske** [rxhanson/Rectangle#690] [Ideas]
I'd like Rectangle to be able to bring a window (i.e. App's main window) to the current Desktop.

My particular use case is for apps which have only one window (they don't support multi-window). I want to use them in the Desktop I'm currently in but then I have to switch to the Desktop where the App is and manually drag it to the Desktop I want them to be (or just use them on the other Desktop, which sometimes is annoying).

I would like to know if having something like this is desired for Rectangle or if there's something already in the roadmap. I make myself available to implement the feature as I'm a programmer (although I'm not familiar with macOS ecosystem). But I don't want to implement a feature in Rectangle only for it to be rejected into the mainline repo :)

**Comment by rxhanson** [comment]
https://github.com/rxhanson/Rectangle#rectangle-doesnt-have-the-ability-to-move-to-other-desktopsspaces

I'm ok with merging a **good** pull request for this, but I won't add this functionality into the UI. I tried the private API a while back and it didn't work. The way Amethyst does this is pretty fragile in my opinion, and that code would have to be rewritten entirely to fit with Rectangle's coding style and remove unnecessary parts. In addition to being a hidden config for this, it would come with the disclaimer that I would not be maintaining any of this code.

**Reply by andrepiske** [reply]
Oh, sorry that I didn't see this already in the readme. And thanks for the link pointing out the issue!
It totally makes sense, it'd just add a burden to maintain code that uses a non-public API. I wouldn't do it either.

Consider this discussion closed (as I don't see any "close discussion" button anywhere).
